### ISSUE TITLE
introduce interface for data fetching

### ISSUE THREAD

**Issue by tonybart1337** [bilibili/web-demuxer#24]
In order to use this library in our application we would need a way to fetch data from a remote.
I like the way you patched it [here](https://github.com/bilibili/web-demuxer/blob/main/lib/web-demuxer/post.js#L64), which means that it should be relatively easy to do

I would like to propose some kind of interface that a user will be able to pass to the instance of a demuxer, so when we call `load` or any other function it will use that interface to fetch the data. Caching can also be implemented into the interface as well

**Comment by ForeverSc** [comment]
Yep, I have the same idea. This has been discussed at https://github.com/bilibili/web-demuxer/issues/11#issuecomment-2421209165. I need some spare time to design how to add the interface. If you have better ideas, you’re welcome to submit a pull request. :)

**Comment by parthibx24** [comment]
Hi, any update on this? whats the best approach to make web-demuxer read and seek the file from a custom data source (eg datachannel)?

**Comment by ForeverSc** [comment]
> Hi, any update on this? whats the best approach to make web-demuxer read and seek the file from a custom data source (eg datachannel)?

Currently, I don't have enough time to redesign this part. If necessary, the file reading method can be customized, and you may refer to the code for [URL reading](https://github.com/bilibili/web-demuxer/blob/main/lib/web-demuxer/post.js#L64-L83)
